funding/manager.go

```diff
@@ -152,13 +152,32 @@ func (m *Manager) deriveFundingShim(ourOrder order.Order,
 
 		thawHeight      uint32
 		selfChanBalance btcutil.Amount
+		ourMultiSigKey  *keychain.KeyDescriptor
 	)
-	if ourOrder.Type() == order.TypeBid {
+	ourOrderBid, ourOrderIsBid := ourOrder.(*order.Bid)
+	ourOrderIsSidecar := ourOrderIsBid &&
+		ourOrderBid.SidecarTicket != nil
+
+	if ourOrderIsBid {
 		bidNonce = ourOrder.Nonce()
 		askNonce = matchedOrder.Order.Nonce()
 
-		thawHeight = ourOrder.(*order.Bid).LeaseDuration
-		selfChanBalance = ourOrder.(*order.Bid).SelfChanBalance
+		thawHeight = ourOrderBid.LeaseDuration
+		selfChanBalance = ourOrderBid.SelfChanBalance
+
+		// If our order is a sidecar channel, we already have the
+		// multisig key in the ticket and don't need to re-derive it
+		// again here.
+		ticket := ourOrderBid.SidecarTicket
+		if ourOrderIsSidecar && ticket.Recipient != nil {
+			ourMultiSigKey = &keychain.KeyDescriptor{
+				KeyLocator: keychain.KeyLocator{
+					Family: keychain.KeyFamilyMultiSig,
+					Index:  ticket.Recipient.MultiSigKeyIndex,
```

Comment:
Does the locator information actually need to be set if we just have their raw pubkey? Same goes for the index within the ticket itself.

Reply:
Unfortunately, we need both the public key as well as the locator (index) for the 3-way authentication in the auctioneer client. The `SignMessage` RPC only takes a locator so we need to know the index. And in fact, looking at the use of `ChanPointShim` in `lnd`, it looks like we only ever use the locator part there as well and ignore the pubkey itself.